py/torch_tensorrt/dynamo/conversion/_conversion.py

```diff
@@ -17,20 +17,24 @@
     TRTInterpreterResult,
 )
 from torch_tensorrt.dynamo.runtime import PythonTorchTensorRTModule, TorchTensorRTModule
-from torch_tensorrt.dynamo.utils import get_model_device, get_torch_inputs
+from torch_tensorrt.dynamo.utils import (
+    get_model_device,
+    get_torch_inputs,
+    unwrap_tensor_shape,
+)
 
 logger = logging.getLogger(__name__)
 
 
-def infer_module_output_dtypes(
+def infer_module_output_shapes_dtypes(
     module: torch.fx.GraphModule,
     inputs: Sequence[Input],
     device: Device,
     kwarg_inputs: Optional[dict[str, Any]] = None,
     truncate_double: bool = False,
-) -> List[dtype]:
+) -> Tuple[List[Tuple[int]], List[dtype]]:
```

Comment:
This probably should be some sort of struct very similar to Input, maybe we can use a fake tensor or something?

Reply:
@peri044 can we replace this with graph inspection?